o2a/mappers/fs_mapper.py

```diff
@@ -112,47 +116,79 @@ class FsMapper(ActionMapper):
     Converts a FS Oozie node to an Airflow task.
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(
+        self,
+        oozie_node: Element,
+        name: str,
+        dag_name: str,
+        property_set: PropertySet,
+        trigger_rule: str = TriggerRule.ALL_SUCCESS,
+        **kwargs,
+    ):
+        super().__init__(
+            oozie_node=oozie_node,
+            name=name,
+            property_set=property_set,
+            trigger_rule=trigger_rule,
+            dag_name=dag_name,
+            **kwargs,
+        )
+        self.oozie_node = oozie_node
+        self.trigger_rule = trigger_rule
         self.tasks: List[Task] = []
 
     def on_parse_node(self):
         super().on_parse_node()
         self.tasks = self.parse_tasks()
 
-    def parse_tasks(self):
+    def parse_tasks(self) -> List[Task]:
         if not list(self.oozie_node):
             return [Task(task_id=self.name, template_name="dummy.tpl", trigger_rule=self.trigger_rule)]
 
-        return [self.parse_fs_action(i, node) for i, node in enumerate(self.oozie_node)]
-
-    def to_tasks_and_relations(self):
+        tasks: List[Task] = []
+        index = 0
+        for node in self.oozie_node:
+            task = self.parse_fs_action(index, node)
+            if task is not None:
+                tasks.append(task)
+                index += 1
+            else:
+                # configuration node needs to be skipped
+                pass
+        return tasks
+
+    def to_tasks_and_relations(self) -> Tuple[List[Task], List[Relation]]:
         return self.tasks, chain(self.tasks)
 
     def required_imports(self) -> Set[str]:
-        return {
-            "import shlex",
-            "from airflow.operators import dummy_operator",
-            "from airflow.operators import bash_operator",
-        }
+        return {"from airflow.operators import dummy_operator", "from airflow.operators import bash_operator"}
 
     @property
-    def first_task_id(self):
+    def first_task_id(self) -> str:
         return self.tasks[0].task_id
 
     @property
-    def last_task_id(self):
+    def last_task_id(self) -> str:
         return self.tasks[-1].task_id
 
-    def parse_fs_action(self, index: int, node: Element):
+    def parse_fs_action(self, index: int, node: Element) -> Optional[Task]:
         tag_name = node.tag
+        if tag_name not in FS_OPERATION_MAPPERS.keys():
+            # Skip nodes that are not FS mapper action nodes
+            return None
         tasks_count = len(self.oozie_node)
```

Comment:
This assumption is now incorrect. Nodes can be more than operations.

Reply:
My patch is available: 
https://github.com/GoogleCloudPlatform/oozie-to-airflow/commit/d2691556b3c8b5840b626c813b6c474277c17ba1

Reply:
I left the discussion above open.